### ISSUE TITLE
arm64 builds

### ISSUE THREAD

**Issue by tomjn** [mailhog/MailHog#359]
Currently `linux_arm` builds are available, but Apple Silicon `arm64` builds are not.

**Comment by evertiro** [comment]
To elaborate on the above, we use `MailHog` and `mhsendmail` in VVV. We actually don't use an `arm64` build currently, but may well in the future. We currently use an `aarch64` build (which is also not provided). I'm currently providing downloads for both `MailHog` and `mhsendmail` on both `arm64` and `aarch64` oh my forks.

https://github.com/evertiro/mhsendmail/releases/tag/v0.2.0-M1

https://github.com/evertiro/MailHog/releases/tag/v1.0.0-M1

**Comment by abbluiz** [comment]
@evertiro what about docker though?

**Comment by tomjn** [comment]
@abbluiz are you suggesting we use docker + QEMU x86 emulation? What about Raspberry Pis, arm servers, or native Apple Silicon?

**Comment by abbluiz** [comment]
@tomjn no, I was just asking really... because there's no arm docker images for mailhog and that would be good, and it would on apple m1 & other arm-based docker hosts

**Comment by ojtibi** [comment]
@abbluiz Exactly. I came here from Docker Hub, and I found this is the only ticket related to AArch64/Apple Silicon. Will be subscribing to notifications on this dicussion.

**Comment by websafe** [comment]
Would be great to have an official arm64 build for M1 based machines.
👍

**Comment by jcalonso** [comment]
For those looking for an `arm64`/`aarch64` Docker image, I built one here: 
https://hub.docker.com/r/jcalonso/mailhog

There is also an arm64 binary here: 
https://github.com/jcalonso/MailHog/releases/tag/v1.0.1-arm 

The arm build was tested on AWS Graviton t4g and Raspberry pi, I don't have a mac to test there but it _should work™_.

It would be great to have this on the official repo, let me know if the releases work fine and I can open an PR.

Edit: Fixed Docker Hub link

**Comment by mikescott** [comment]
@jcalonso your binary works perfectly inside a Debian 11 arm64 VM on an Apple Silicon Mac

**Comment by Enchiridion** [comment]
@jcalonso Thanks for making that version!  Working on Ubuntu 22 on M1 Mac.  Any chance you could also make an arm64 version of tkotosz/mhsendmail and tkotosz/mhsendmail-wrapper?

**Comment by mikescott** [comment]
I discovered it's a piece of cake to compile these:

1. Install `golang-go`
2. Clone MailHog or mhsendmail
3. `cd` to the directory and run `make` and run the command it suggests to fix the errors, and run `make` again
4. The binaries should be in `~/go/bin`

**Comment by atezet** [comment]
I LITERALLY just forked the image and ran 
```
docker buildx build --push --platform linux/arm64,linux/amd64 -t arjenz/mailhog:latest .
```
to build and push it to Docker Hub. You're free to use it: https://hub.docker.com/r/arjenz/mailhog

If you want to try out a new project, created by a colleague of mine (in Rust), here's MailCrab: https://github.com/tweedegolf/mailcrab ! Also available on Docker Hub: https://hub.docker.com/r/marlonb/mailcrab

**Comment by extralooping** [comment]
> I discovered it's a piece of cake to compile these:
> 
>     1. Install `golang-go`
> 
>     2. Clone MailHog or mhsendmail
> 
>     3. `cd` to the directory and run `make` and run the command it suggests to fix the errors, and run `make` again
> 
>     4. The binaries should be in `~/go/bin`

For me (debian 11 inside parallelsvm on m3 mac), it was even easier with `go get github.com/mailhog/MailHog`, the result was in `~/go/bin`
